improver/blending/weighted_blend.py

```diff
@@ -902,8 +913,9 @@ def process(self, cube, weights=None, cycletime=None, attributes_dict=None):
             result = self.weighted_mean(cube, weights)
         self._update_blended_metadata(result, attributes_dict)
 
-        # Re-mask output
-        if isinstance(cube.data, np.ma.core.MaskedArray):
-            result.data = np.ma.array(result.data)
+        # Checks the coordinate dimensions match the first relevant cube in the unblended cubeList.
+        index = [slice(None)] * cube.ndim
+        index[cube.coord_dims(self.blend_coord)[0]] = 0
+        result = check_cube_coordinates(cube[tuple(index)], result)
```

Comment:
instead of `cube[tuple(index)]` you could use `next(cube.slices_over(self.blend_coord))`